### ISSUE TITLE
Fragile parsing of anchor references

### ISSUE THREAD

**Issue by vdukhovni** [haskell/haddock#1284]
When an anchor reference appears as the first thing on a line it seems to render verbatim, and one needs to be sure to prepend some other word (reflow the text) to avoid this:
```
bad:
--  ... blah, blah
-- [strict](#strict) blah

good:
-- ... blah,
-- blah [strict](#strict) blah
```

**Comment by sjakobi** [comment]
Possible duplicate of https://github.com/haskell/haddock/issues/774.

**Comment by vdukhovni** [comment]
Agreed, looks to be the same. Is there a fix in the pipeline?

**Comment by vdukhovni** [comment]
I can close this issue if #774 is not forever dormant...

**Comment by sjakobi** [comment]
> Is there a fix in the pipeline?

I don't think so, unfortunately. Apparently I didn't make a lot of progress when I initially looked into #774, and I'm not planning on returning to that issue any time soon.
